Source/Core/Core/ConfigManager.cpp

```diff
@@ -639,6 +639,17 @@ void SConfig::LoadJitDebugSettings(IniFile& ini)
   section->Get("JitRegisterCacheOff", &bJITRegisterCacheOff, false);
 }
 
+void SConfig::LoadSIDevice0Setting()
+{
+  Config::Load();
+  IniFile ini;
+  ini.Load(File::GetUserPath(F_DOLPHINCONFIG_IDX));
+  IniFile::Section* core = ini.GetOrCreateSection("Core");
+
+  core->Get("SIDevice0", &m_SIDevice[0],
+            0 ? SerialInterface::SIDEVICE_NONE : SerialInterface::SIDEVICE_GC_CONTROLLER);
```

Comment:
Is there any good reason why this should load from the INI instead of using a value passed in as a parameter? What's especially strange is that this is called from a function which does receive the right value as a parameter as far as I can tell.

Also, `0 ? SerialInterface::SIDEVICE_NONE : SerialInterface::SIDEVICE_GC_CONTROLLER` is just a more complicated way of writing `SerialInterface::SIDEVICE_GC_CONTROLLER`.

Reply:
INI changes to `SIDevice0` are made before `NativeLibrary.SetOverlaySIDevice` is called. Removing `SConfig::GetInstance().LoadSIDevice0Setting()` caused Dolphin to fail to disable the overlay GC controller with "Disable Unselected Wii Overlay Controllers" enabled. I suspect`SerialInterface::ChangeDevice` will eventually read `SIDevice0`'s value, which means it needs to be reloaded beforehand.

And I used your `SerialInterface::SIDEVICE_GC_CONTROLLER` suggestion.

Reply:
Yes, I understand that you have to call `LoadSIDevice0Setting`, but I don't understand why `LoadSIDevice0Setting` has to load data from the INI file instead of receiving that data as a parameter.

Reply:
So I should pass in `gcController` and change `core->Get` to `core->Set`?

Reply:
For some reason I didn't initially think of `core->Set`. Since we're setting one specific setting, let's just do that in this method.